Modified github.md : Added how to create a viable forked repo (#19)
introcompsys/fall2022: faq/github.md

````
Doing each new assignment in its own branch is best practice. In a typical software development flow once the codebase is stable a new branch would be created for each new feature or patch. This analogy should help you build intuition for this GitHub flow and using branches.  Also, pull requests are the best way for us to give you feedback. Also, if you create a branch when you do not need it, you can easily merge them after you are done, but it is hard to isolate things onto a branch if it's on main already.
````

## How would I continue to add to this FAQ page by myself?
````{toggle}

> Getting everything set up is really simple! Though it gets a little bit technical. This will describe the process to fork any remote repo and allow you to use ```git fetch``` to get information from the remote repo you created the fork from and ```git push``` to your fork (which can then be merged by the original owner you forked from).

Reviewer: ```suggestion
Getting everything set up is really simple! Though it gets a little bit technical. This will describe the process to fork any remote repo and allow you to use `git fetch` to get information from the remote repo you created the fork from and `git push` to your fork (which can then be merged by the original owner you forked from).
```

inline uses one backtick, 3 is for blocks

This is useful for suggesting changes to a repository that you do not own or have access to at all, with the original owner still being able to accept or deny the changes.

On the GitHub repository you wish to fork, click the "Fork" icon at the top of the page. You can then rename the repo, only copy the main branch etc., however we just want to continue.

It then brings you to the new remote repository that is now under your name.

> Clone the repo using ```git clone <clone-link>``` in your directory of choosing.

Reviewer: ```suggestion
Clone the repo using `git clone <clone-link>` in your local directory of choice.
```
minor fixes

> NOTE: "upstream" in the above command can be replaced with whatever you want, just note that you will use that name when using other commands referencing it.

When using ```git remote -v``` again, you now get the following back:

> origin <https://github.com/<username>/<repo-name>.git> (fetch)
> origin <https://github.com/<username>/<repo-name>.git> (push)
> upstream <https://github.com/<original-owners-username>/<repo-name>.git> (fetch)
> upstream <https://github.com/<original-owners-username>/<repo-name>.git> (push)

You can now run the command 
```
git fetch upstream
```
to successfully fetch all information from the original remote repo.

Running 
```
git merge upstream/main
```
then merges the fetched information from the remote repo into your local forked repo.

When pushing your changes, you will continue to either use
```
git push
```
or
```
git push origin
```
of course after pushing upstream for the first time. 

Remember, you are not pushing to the original repo, you are pushing to your fork.  You can only pull and fetch from the original remote repo.

> I hope this helps!

Reviewer: ```suggestion

```
